apps/examples/testcase/ta_tc/mqtt/itc/itc_mqtt_main.c

```diff
@@ -275,6 +388,429 @@ void itc_mqtt_subscribe_unsubscribe_p(void)
+	ITC_MQTT_WAIT_SIGNAL;
+
+	res = mqtt_subscribe(g_mqtt_client_handle, ITC_MQTT_TOPIC, 0);
+	TC_ASSERT_EQ_CLEANUP("mqtt_subscribe", res, 0,
+		mqtt_disconnect(g_mqtt_client_handle); mqtt_deinit_client(g_mqtt_client_handle));
+
+	res = mqtt_unsubscribe(g_mqtt_client_handle, ITC_MQTT_TOPIC);
+	TC_ASSERT_EQ_CLEANUP("mqtt_unsubscribe", res, 0,
+		mqtt_disconnect(g_mqtt_client_handle); mqtt_deinit_client(g_mqtt_client_handle));
+
+	//MQTT re-unsubscribe
+	res = mqtt_unsubscribe(g_mqtt_client_handle, ITC_MQTT_TOPIC);
+	TC_ASSERT_EQ_CLEANUP("mqtt_unsubscribe", res, 0,
+		mqtt_disconnect(g_mqtt_client_handle); mqtt_deinit_client(g_mqtt_client_handle));
+
+	res = mqtt_disconnect(g_mqtt_client_handle);
+	TC_ASSERT_EQ_CLEANUP("mqtt_disconnect", res, 0, mqtt_deinit_client(g_mqtt_client_handle));
+	ITC_MQTT_WAIT_SIGNAL;
+
+	res = mqtt_deinit_client(g_mqtt_client_handle);
+	TC_ASSERT_EQ("mqtt_deinit_client", res, 0);
+	TC_SUCCESS_RESULT();
+}
+
+/*
+* @testcase         itc_mqtt_api_success_ratio
+* @brief            success ratio for all api
+* @scenario         success ratio for all api
+* @apicovered       none
+* @precondition     none
+* @postcondition    none
+*/
+
+void itc_mqtt_api_success_ratio(void)
+{
+	int res;
+	int i;
+	float init_cnt = 0.0f;
+	float s_init_cnt = 0.0f;
+	float deinit_cnt = 0.0f;
+	float s_deinit_cnt = 0.0f;
+	float conn_cnt = 0.0f;
+	float s_conn_cnt = 0.0f;
+	float disconn_cnt = 0.0f;
+	float s_disconn_cnt = 0.0f;
+	float subs_cnt = 0.0f;
+	float s_subs_cnt = 0.0f;
+	float unsubs_cnt = 0.0f;
+	float s_unsubs_cnt = 0.0f;
+	float publish_cnt = 0.0f;
+	float s_publish_cnt = 0.0f;
+
+	for (i = 0; i < ITC_MQTT_LOOP_SIZE; i++) {
+		g_mqtt_client_handle = mqtt_init_client(&g_mqtt_client_config);
+		if (g_mqtt_client_handle != NULL) {
+			s_init_cnt++;
+		} else {
+			init_cnt++;
+			continue;
+		}
+		init_cnt++;
+
+		res = mqtt_connect(g_mqtt_client_handle, CONFIG_EXAMPLES_TESTCASE_ITC_MQTT_BROKER_ADDR,	CONFIG_EXAMPLES_TESTCASE_ITC_MQTT_BROKER_PORT, 0);
+		if (res == 0) {
+			s_conn_cnt++;
+		} else {
+			conn_cnt++;
+			res = mqtt_deinit_client(g_mqtt_client_handle);
+			if (res != 0) {
+				printf("mqtt_deinit_client failed\n");
+			}
+			continue;
+		}
+		conn_cnt++;
+		ITC_MQTT_WAIT_SIGNAL;
+
+		res = mqtt_subscribe(g_mqtt_client_handle, ITC_MQTT_TOPIC, 0);
+		if (res == 0) {
+			s_subs_cnt++;
+		}
+		subs_cnt++;
+
+		res = mqtt_publish(g_mqtt_client_handle, ITC_MQTT_TOPIC, g_mqtt_msg,
+				sizeof(g_mqtt_msg), 0, 0);
+		if (res == 0) {
+			s_publish_cnt++;
+		}
+		publish_cnt++;
+		ITC_MQTT_WAIT_SIGNAL;
+
+		res = mqtt_unsubscribe(g_mqtt_client_handle, ITC_MQTT_TOPIC);
+		if (res == 0) {
+			s_unsubs_cnt++;
+		}
+		unsubs_cnt++;
+
+		res = mqtt_disconnect(g_mqtt_client_handle);
+		if (res == 0) {
+			s_disconn_cnt++;
+		}
+		disconn_cnt++;
+		ITC_MQTT_WAIT_SIGNAL;
+
+		res = mqtt_deinit_client(g_mqtt_client_handle);
+		if (res == 0) {
+			s_deinit_cnt++;
+		}
+		deinit_cnt++;
+	}
+	printf("\n\t---Success Rate---\n");
```

Comment:
it is safer to check for divide by zero cases here.

Reply:
TC_ASSERT validate and take care divisible by zero. In worst scenario initialization is getting failed all the time; in that case s_init_cnt and init_cnt will not be equal so system mark as failed testcase and return; system will not check further; hence there no scenario in which condition will raised where any value will be divisible by Zero.

"TC_ASSERT_EQ("init sucesses rate", s_init_cnt, init_cnt);"

Reply:
1. if init fails all the time, both s_conn_cnt and conn_cnt stay 0, because of the continue statement that avoids the rest of the loop. In the end, there is a 0/0 division case here. Please check and ignore if not the case.

2. TC_ASSERT_EQ needs to be done before printf, yet, here it is done following each printf. Please check and ignore if not the case.